compiler/rustc_codegen_ssa/src/back/archive.rs

```diff
@@ -146,17 +144,23 @@ pub fn try_extract_macho_fat_archive(
         _ => return Ok(None),
     };
 
-    match object::macho::FatHeader::parse(&*archive_map) {
-        Ok(h) if h.magic.get(object::endian::BigEndian) == object::macho::FAT_MAGIC => {
-            let archs = object::macho::FatHeader::parse_arch32(&*archive_map);
+    if let Ok(h) = object::read::macho::MachOFatFile32::parse(&*archive_map) {
+        if h.header().magic.get(object::endian::BigEndian) == object::macho::FAT_MAGIC {
```

Comment:
This check is already done at https://github.com/gimli-rs/object/blob/12d0948a288e38f2458bc5d99f8fea2405224424/src/read/macho/fat.rs#L37

Reply:
https://github.com/rust-lang/rust/pull/127860/commits/616899dcc5abe89a9dd0fc3e3b00e1b2d95e995c